### PR TITLE
[INLONG-5224][Agent][DataProxy] Remove unused StreamConfigLog related classes

### LINKED ISSUE #5224: [Improve][Agent][DataProxy] Remove unused StreamConfigLog related classes
### Description

Remove unused StreamConfigLog and its related classes.

### InLong Component

InLong Manager

### Are you willing to submit PR?

- [X] Yes, I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by healchow**
### Prepare a Pull Request

- Fixes #5224

### Motivation

Remove unused StreamConfigLog related classes.

### Verifying this change

- [x] This change is a trivial rework/code cleanup without any test coverage.

### CODE REVIEW

File: inlong-agent/agent-plugins/src/main/java/org/apache/inlong/agent/plugin/sources/reader/BinlogReader.java [hunk 1]
```diff
@@ -150,84 +141,54 @@ public void init(JobProfile jobConf) {
...
 
-        enableReportConfigLog =
-                Boolean.parseBoolean(jobConf.get(StreamConfigLogMetric.CONFIG_LOG_REPORT_ENABLE,
-                        "true"));
-
-        inlongGroupId = jobConf.get(CommonConstants.PROXY_INLONG_GROUP_ID,
-                CommonConstants.DEFAULT_PROXY_INLONG_GROUP_ID);
-        inlongStreamId = jobConf.get(CommonConstants.PROXY_INLONG_STREAM_ID,
-                CommonConstants.DEFAULT_PROXY_INLONG_STREAM_ID);
         metricTagName = BINLOG_READER_TAG_NAME + "_" + inlongGroupId + "_" + inlongStreamId;
```
Comment (haibo-duan) [inline inlong-agent/agent-plugins/src/main/java/org/apache/inlong/agent/plugin/sources/reader/BinlogReader.java:161]: Use string.join may be better.

Comment (healchow) [reply inlong-agent/agent-plugins/src/main/java/org/apache/inlong/agent/plugin/sources/reader/BinlogReader.java]: done.

**Review by dockerzhang — APPROVED** [review]

**Review by haibo-duan — APPROVED** [review]

**Review by pocozh — APPROVED** [review]
LGTM

**Review by baomingyu — APPROVED** [review]
+1

**Review by EMsnap — APPROVED** [review]

**Pull request merged**